File: Set-Backend/Models/Card.cs
namespace Set_Backend.Models;

public enum CardColour { Red, Green, Purple}
public enum CardShape { Diamond, Squiggle, Oval }
public enum CardFilling { Solid, Striped, Open }
public enum CardNumber { One = 1, Two = 2, Three = 3 }

public class Card
{
    public required int Id { get; set; }
    
    public CardColour Colour { get; set; }
    
    public CardShape Shape { get; set; }
    
    public CardFilling Filling { get; set; }
    
    public CardNumber Number { get; set; }
    
}

File: Set-Frontend/src/app/models/card.ts
export enum CardColour { Red = 'Red', Green = 'Green', Purple = 'Purple' }
export enum CardShape { Diamond = 'Diamond', Squiggle = 'Squiggle', Oval = 'Oval' }
export enum CardFilling { Solid = 'Solid', Striped = 'Striped', Open = 'Open' }
export enum CardNumber { One = 'One', Two = 'Two', Three = 'Three' }

export interface Card {
  id: number;
  colour: CardColour;
  shape: CardShape;
  filling: CardFilling;
  number: CardNumber;
  selected?: boolean;
  hinted?: boolean;
}
